### PR TITLE
Added the database functions

### PR DESCRIPTION
**Pull request by ozzy1323232**
Documentation listed on discord

- will need deletion soon
- Needs feedback on what extra functions are needed. At the moment only table Attendance can be updated

### CODE REVIEW

File: app/database.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+# sql
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import and_
+from sqlalchemy.exc import IntegrityError
+
+def SignOut(studentID, sessionID):
+    
+    ## IMPORTANT
+    # this may have a glitch where there are multiple rows with the same sessionID
+    # and studentID which will confuse the query, for now im hoping the first() will naturally pick the most recent sessionID row with a blank signoutTime column instead of a row from weeks ago
```
Comment (DeltaO3) [inline app/database.py:17]: Could you not just add a check for having blank signoutTime to the query then?

File: app/database.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            consent         = consent)            
+
+        db.session.add(StudentEntry)    # add the changes 
+        db.session.commit()             # save the changes
+    
+    except IntegrityError as e:
+        db.session.rollback()
+        print(f'An error occurred: {e}')  
+
+    return 0
```
Comment (DeltaO3) [inline app/database.py:53]: Dont believe the return statement is necessary here, as well as other times when we are not wanting any data further down the file

File: app/database.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+    
+    attendance_records = query.all()
+    
+    # Debugging, this is how you access the records 
+    for record in attendance_records:
+        
+        print(f"Attendance ID: {record.attendanceID}, Sign-In Time: {record.signInTime}, Sign-Out Time: {record.signOutTime}")
+        
+    return 0
```
Comment (DeltaO3) [inline app/database.py:171]: Should be returning the queried object. in this case, return attendance_records. 

Similarly for most get functions further down

**Review by DeltaO3 — CHANGES_REQUESTED** [review]
Overall, looks fine, I think that adding error checking is quite a good idea and one I hadn't thought of.
Some minor issues as pointed out in the below review comments:
get functions should be returning the requested objects, so we can actually use the functions as intended.
Non get functions _probably_ dont need a return statement.
You should attempt to avoid assumptions about the database imo, can you not query for an empty/null entry in signoutTime?

File: app/database.py (hunk 3 above)
Comment (ozzy1323232) [reply app/database.py:171]: sorry i forgot that, will fix now

**Review by DeltaO3 — APPROVED** [review]
LGTM!

**Review by Ay1tsMe — APPROVED** [review]

**Pull request merged**